src/components/Web3Modal.tsx

```diff
@@ -95,6 +106,10 @@ export function Web3Modal({
           ClientCtrl.setProvider(provider);
           provider.on('display_uri', onDisplayUri);
           provider.on('session_delete', onSessionDelete);
+          provider.client.core.relayer.subscriber.on(
```

Comment:
Why is this needed in addition to above `provider.on('session_delete', onSessionDelete);` ?

Reply:
I need this one to cover the case where 'provider.disconnect' is called from outside the sdk. It's a possible case as im exposing the provider in useWeb3Modal hook.

I could remove the "session_delete" listener and use just the core one.

Reply:
Do you reckon using `provider.client` approach would cover all of the cases in general? If so, could just use that I guess

Reply:
yes, i tested disconnecting from outside the sdk, and from the wallet side and they both work. Is there other possible case?